spec/dynamoid/persistence_spec.rb

```diff
@@ -559,6 +560,13 @@ def reload_address
         expect(v).to include(s)
       }
     end
+
+    it 'does not load parent item when quering the child table' do
+      vehicle && car
+
+      expect(Car.all).to contain_exactly(car)
+      expect(Car.all).not_to include(vehicle)
```

Comment:
Also make sure this doesn't load `sub` in the set too since different type.

Reply:
:+1: will update PR on Monday